pkg/metadata/store/metadata_store_test.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package store
+
+import (
+	"bytes"
+	"errors"
+	"testing"
+
+	assertlib "github.com/stretchr/testify/assert"
+)
+
+func TestMetadata(t *testing.T) {
+	testData := [][]byte{
+		[]byte("test-data-1"),
+		[]byte("test-data-2"),
+	}
+	updateErr := errors.New("update error")
+	assert := assertlib.New(t)
+
+	t.Logf("simple create and get")
+	meta, err := newMetadata(testData[0])
+	assert.NoError(err)
+	old := meta.get()
+	assert.Equal(testData[0], old)
+
+	t.Logf("failed update should not take effect")
+	err = meta.update(func(in []byte) ([]byte, error) {
+		return testData[1], updateErr
+	})
+	assert.Equal(updateErr, err)
+	assert.Equal(testData[0], meta.get())
+
+	t.Logf("successful update should take effect")
+	err = meta.update(func(in []byte) ([]byte, error) {
+		return testData[1], nil
+	})
+	assert.NoError(err)
+	assert.Equal(testData[1], meta.get())
+
+	t.Logf("successful update should not affect existing snapshot")
+	assert.Equal(testData[0], old)
+
+	// TODO(random-liu): Test deleteCheckpoint and cleanupCheckpoint after
+	// disk based implementation is added.
+}
+
+func TestMetadataStore(t *testing.T) {
+	testIds := []string{"id-0", "id-1"}
+	testMeta := map[string][]byte{
+		testIds[0]: []byte("metadata-0"),
+		testIds[1]: []byte("metadata-1"),
+	}
+	assert := assertlib.New(t)
+
+	m := NewMetadataStore()
+
+	t.Logf("should be empty initially")
+	metas, err := m.List()
+	assert.NoError(err)
+	assert.Empty(metas)
```

Comment:
I just replace the comment with logging, or else still need to comment for each test case. :)